src/System.Private.CoreLib/shared/System/Globalization/TimeSpanFormat.cs

```diff
@@ -35,66 +37,53 @@ private static unsafe void AppendNonNegativeInt32(StringBuilder sb, int n, int d
         internal static readonly FormatLiterals PositiveInvariantFormatLiterals = TimeSpanFormat.FormatLiterals.InitInvariant(isNegative: false);
         internal static readonly FormatLiterals NegativeInvariantFormatLiterals = TimeSpanFormat.FormatLiterals.InitInvariant(isNegative: true);
 
-        internal enum Pattern
-        {
-            None = 0,
-            Minimum = 1,
-            Full = 2,
-        }
 
         /// <summary>Main method called from TimeSpan.ToString.</summary>
-        internal static string Format(TimeSpan value, string format, IFormatProvider formatProvider) =>
-            StringBuilderCache.GetStringAndRelease(FormatToBuilder(value, format, formatProvider));
+        internal static string Format(TimeSpan value, string format, IFormatProvider formatProvider)
+        {
+            return IsFormatC(format) ? // special-case to optimize the default TimeSpan format
+                FormatC(value) : // formatProvider ignored, as "c" is invariant
+                StringBuilderCache.GetStringAndRelease(FormatToBuilder(value, format, formatProvider));
+        }
 
         /// <summary>Main method called from TimeSpan.TryFormat.</summary>
         internal static bool TryFormat(TimeSpan value, Span<char> destination, out int charsWritten, ReadOnlySpan<char> format, IFormatProvider formatProvider)
         {
+            if (IsFormatC(format)) // special-case to optimize the default TimeSpan format
+            {
+                return TryFormatC(value, destination, out charsWritten); // formatProvider ignored, as "c" is invariant
+            }
+
             StringBuilder sb = FormatToBuilder(value, format, formatProvider);
+
             if (sb.Length <= destination.Length)
             {
-                charsWritten = sb.Length;
                 sb.CopyTo(0, destination, sb.Length);
+                charsWritten = sb.Length;
                 StringBuilderCache.Release(sb);
                 return true;
             }
-            else
-            {
-                StringBuilderCache.Release(sb);
-                charsWritten = 0;
-                return false;
-            }
+
+            charsWritten = 0;
+            StringBuilderCache.Release(sb);
+            return false;
         }
 
         private static StringBuilder FormatToBuilder(TimeSpan value, ReadOnlySpan<char> format, IFormatProvider formatProvider)
         {
-            if (format.Length == 0)
-            {
-                format = "c";
-            }
-
-            // Standard formats
+            // Standard formats other than 'c'/'t'/'T', which should have already been handled.
```

Comment:
Debug.Assert?

Reply:
We throw below.  That seemed sufficient.